store/src/java/com/zimbra/cs/filter/jsieve/EditHeaderExtension.java

```diff
@@ -564,4 +566,35 @@ private boolean matchValue(String regex, String value) {
         Matcher matcher = pattern.matcher(value);
         return matcher.matches();
     }
+
+    static public boolean isTolerableFormedMessage(ZimbraMailAdapter zma, String actionName, MimeMessage mm) {
```

Comment:
saveChanges is called twice now, once during malformed mime check and again after editing the mime. instead can't we just call the save changes after editing the mime as it was done earlier and just catch the exceptions like it is being done in saveChanges() method below.